### ISSUE TITLE
Why is training so slow？

### ISSUE THREAD

**Issue by YeYQ22** [milesial/Pytorch-UNet#457]
I set the training batchsize to 4, a total of about 4,500 pieces of training data, according to the speed of training, training rounds take about 10 hours or so, this speed is unacceptable, I use a graphics card is 2080Ti, the parameters of the configuration as shown in Figure：
![image](https://github.com/milesial/Pytorch-UNet/assets/118253577/c1ec4592-e3b0-40f0-9a10-772abdde9916)
![image](https://github.com/milesial/Pytorch-UNet/assets/118253577/60240b1e-ff6b-4ec0-af24-07c6680ec97a)

**Comment by WenBingo** [comment]
hi , I want to know how to down the dataset. I run the code of "bash scripts/download_data.sh", but need the “Kaggle username”and " Kaggle username".  
Please, i don't know how to solve this.

**Comment by IshanG97** [comment]
@WenBingo maybe you figured this out already, but for others: you need to create an account with Kaggle, go to https://www.kaggle.com/competitions/carvana-image-masking-challenge/overview, and click Late Submission, accept their terms and then it should work.

**Comment by sunmianmian123** [comment]
Hi, I have the same problem and the same computer configuration with you. Have you solved the problem?

> I set the training batchsize to 4, a total of about 4,500 pieces of training data, according to the speed of training, training rounds take about 10 hours or so, this speed is unacceptable, I use a graphics card is 2080Ti, the parameters of the configuration as shown in Figure： ![image](https://private-user-images.githubusercontent.com/118253577/263139282-c1ec4592-e3b0-40f0-9a10-772abdde9916.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUzMTg1MjIsIm5iZiI6MTcxNTMxODIyMiwicGF0aCI6Ii8xMTgyNTM1NzcvMjYzMTM5MjgyLWMxZWM0NTkyLWUzYjAtNDBmMC05YTEwLTc3MmFiZGRlOTkxNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxMFQwNTE3MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZDY1MTI3NGZjMzdjMTdiNzdlMjVkNmQwZGIyMWUzMzZmODg4NzY4MDQ5MjM0MDExYzViYzc2NGE0NTU5NWY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.FHVuXrmlefVZENhOKjkR6_um5B5RMiFDOjOClfZo0y8) ![image](https://private-user-images.githubusercontent.com/118253577/263139353-60240b1e-ff6b-4ec0-af24-07c6680ec97a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUzMTg1MjIsIm5iZiI6MTcxNTMxODIyMiwicGF0aCI6Ii8xMTgyNTM1NzcvMjYzMTM5MzUzLTYwMjQwYjFlLWZmNmItNGVjMC1hZjI0LTA3YzY2ODBlYzk3YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxMFQwNTE3MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MDc5NzRkNmJhNWIyNWZiYWMwY2NjYzJjNTM5NjU5NmYzZGZkOWUwNDQ1ZDFlMDE4ZDRjYjJlZDQwMDYwOTYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.negEjg2MzSClty9KXCTX8vxthDaYYV_TwPNKp7sBB2g)

Looking forward to your reply.

**Comment by burning-star-x** [comment]
so slow too with 3060ti,why?
use command: python train.py --amp --batch-size 3

**Comment by Terrylululu** [comment]
> so slow too with 3060ti,why? use command: python train.py --amp --batch-size 3

I have the same  problem. do you know how to slove this.
Looking forward to your reply.

> I set the training batchsize to 4, a total of about 4,500 pieces of training data, according to the speed of training, training rounds take about 10 hours or so, this speed is unacceptable, I use a graphics card is 2080Ti, the parameters of the configuration as shown in Figure： ![image](https://private-user-images.githubusercontent.com/118253577/263139282-c1ec4592-e3b0-40f0-9a10-772abdde9916.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4OTkzNjIsIm5iZiI6MTcyNDg5OTA2MiwicGF0aCI6Ii8xMTgyNTM1NzcvMjYzMTM5MjgyLWMxZWM0NTkyLWUzYjAtNDBmMC05YTEwLTc3MmFiZGRlOTkxNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyOVQwMjM3NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYmI2MzVhZjE4NTZhNjRlMWVmMTFmMDFiOTgzZDlhMGFiZGQxNDI4NmJkNTE4NWRlY2U4MWIxMWZkNTIwNTdmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.HBQtYv_8S5nIb6r41IgtH2BdnAvQ7l_oKznKx7gAPo4) ![image](https://private-user-images.githubusercontent.com/118253577/263139353-60240b1e-ff6b-4ec0-af24-07c6680ec97a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4OTkzNjIsIm5iZiI6MTcyNDg5OTA2MiwicGF0aCI6Ii8xMTgyNTM1NzcvMjYzMTM5MzUzLTYwMjQwYjFlLWZmNmItNGVjMC1hZjI0LTA3YzY2ODBlYzk3YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyOVQwMjM3NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZjA4ODhiNWExY2JhM2Y2ZDA0ZWZiYjY4MDA3NzNkMmI0M2M2OWMzZWEyOWNhMTJiODY0ZGNmMjU1NzBjMmJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4Q9gDHiJBnsU9hIQ6Y390Y9gYJSMVowp4leeeuEKVTQ)

**Comment by burning-star-x** [comment]
@terryluzhenfeng  because -amp
